dnf/cli/cli.py

```diff
@@ -803,6 +803,11 @@ def configure(self, args, option_parser=None):
             e = '%s: %s' % (ucd(e.args[1]), repr(e.filename))
             logger.critical(_('Config error: %s'), e)
             sys.exit(1)
+        if opts.destdir is not None:
+            self.base.conf.destdir = opts.destdir
+            if not self.base.conf.downloadonly and opts.command != 'download':
```

Comment:
Awww, I should've had thought about it for longer and reread what I had written. I'm sure that `and` is proper, the or I proposed (and afterwards said it is "fixed" by it) is the logical negation of it, and not equivalent. Finally fixed in 4ab94a9.